### PR TITLE
Timeout

### PR DESCRIPTION
**Pull request by yofukashino**
30 seconds timeout between completing the queue and leaving the VC. wouldn't leave if something is played in the meantime.

### CODE REVIEW

**Comment by Allvaa** [comment]
>wouldn't leave if something is played in the meantime.

it still leaves vc tho

https://user-images.githubusercontent.com/52526068/132991297-609f85df-b092-4de1-bff4-c4c19411f8dc.mp4

**Comment by yofukashino** [comment]
why its not sending the message of leaving?

File: src/structures/MusicHandler.js [hunk 1]
```diff
@@ -78,13 +78,17 @@ module.exports = class MusicHandler {
...
-                    if (this.textChannel) this.textChannel.send({embeds:[util.embed().setDescription("✅ | Queue is empty. Leaving voice channel..")]});
-                    this.reset();
+                    this.player.track = null;
+                    if (this.textChannel) this.textChannel.send({embeds:[util.embed().setDescription("✅ | Queue is empty. Will Leave voice channel..")]});
+                    setTimeout(() => { 
+                            if (!this.queue.length && !this.current && this.player && !this.player.track)  {
+                               if (this.textChannel) this.textChannel.send({embeds:[util.embed().setDescription("✅ | Nothing is Played. Leaving voice channel..")]});                                                     
+                            this.node.leaveChannel(this.guild.id); 
+                            this.reset();      
+                        }, 30000);
```
Comment (Allvaa) [inline src/structures/MusicHandler.js]: make this millisecond adjustable, also if its 0 don't run the setTimeout, just directly execute the code outside setTimeout

**Review by Allvaa — CHANGES_REQUESTED** [review]

File: src/structures/MusicHandler.js [hunk 2]
```diff
@@ -78,13 +78,17 @@ module.exports = class MusicHandler {
...
                 if (this.loop === 1 && !this.shouldSkipCurrent) this.queue.unshift(this.previous);
                 else if (this.loop === 2) this.queue.push(this.previous);
-
                 if (this.shouldSkipCurrent) this.shouldSkipCurrent = false;
-
                 if (!this.queue.length) {
-                    this.node.leaveChannel(this.guild.id);
-                    if (this.textChannel) this.textChannel.send({embeds:[util.embed().setDescription("✅ | Queue is empty. Leaving voice channel..")]});
-                    this.reset();
+                    this.player.track = null;
```
Comment (Allvaa) [inline src/structures/MusicHandler.js]:
```suggestion
```
can be removed I think

**Review by Allvaa — CHANGES_REQUESTED** [review]

**Comment by Allvaa** [comment]
also there is syntax error

**Comment by Allvaa** [comment]
>https://github.com/Allvaa/lavalink-musicbot/pull/148#issuecomment-917646542

this bug still occurs in your latest commit btw

**Comment by yofukashino** [comment]
well where do you want to store the timeout const?

**Comment by Allvaa** [comment]
> well where do you want to store the timeout const?

wherever, but I prefer in the client class

File: src/structures/MusicHandler.js (hunk 1 above)
Comment (yofukashino) [reply src/structures/MusicHandler.js]: add it and also added that the bot will leave vc if playing alone

**Comment by yofukashino** [comment]
well working fine for me though if giving you any errors than its a pity

**Pull request closed without merging**